component/registry.go

```diff
@@ -49,6 +50,9 @@ type Options struct {
 	// by the component; a component must use the same Exports type for its
 	// lifetime.
 	OnStateChange func(e Exports)
+
+	//Registerer allows components to add their own metrics. The register will come pre-wrapped with the component id
+	Registerer Registerer
```

Comment:
Components shouldn't need to call `RegisterComponent` or `UnregisterComponent`, this should just be the normal prometheus.Registerer interface.

Reply:
Tradeoff is the controller will need to do some casting and it's a bit more boilerplate in the components themselves. I could use MustRegister to register multiple metrics but I don't think we should panic in this case.

Reply:
I think it's probably fine to panic IMO, you'd really want to know fast if you registered invalid metrics, otherwise it'll just be hidden and you won't know why it's not showing up